src/include/txn_inline.h

```diff
@@ -666,11 +726,12 @@ __wt_txn_pinned_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *pinned_tsp)
      * pinned. If a checkpoint is starting and we have to use the checkpoint timestamp, we take the
      * minimum of it with the oldest timestamp, which is what we want.
      */
-    WT_ACQUIRE_BARRIER();
     checkpoint_ts = txn_global->checkpoint_timestamp;
```

Comment:
Where's the checkpoint generation read that the comment references? Does the comment need to be updated?

Reply:
Yes, it's need to be updated. The older version back in 4.2 needs to read the checkpoint generation.